feat: add agoric-upgrade-19 (proposal 91)

muhammadahmadasifbhatti opened the pull request:
<!--
Thank you for submitting a pull request!

Please delete the sections below that aren pertinent to your PR
and then fulfill the remaining checklist items before merging.
-->
### Description

This PR adopts governance proposal 91 as per upgrade 19. Created via following instructions [here](https://github.com/Agoric/agoric-sdk/blob/master/a3p-integration/proposals/README.md). 
[Companion agoric-sdk PR](https://github.com/Agoric/agoric-sdk/pull/11233)

When a proposal passes on agoric-3 Mainnet, it should be included in the history that this synthetic image tracks.

- [x] before this PR, change any `fromTag` using `latest` (such as [a3p-integration](https://github.com/Agoric/agoric-sdk/blob/master/a3p-integration/package.json)) to use a fixed version (otherwise they will fail when this PR changes latest and they pick it up)
- [x] before merging this PR, include a link to a PR that adopts its fromTag `use-${proposalName}` (where proposalName is the part of the agoric-3-proposals proposal directory name after the colon, cf. [a3p-integration/proposals](https://github.com/Agoric/agoric-sdk/tree/master/a3p-integration/proposals#how-to-revise-this-directory-after))
- [ ] after this PR merges, merge that other PR

Muneeb147 commented:
@muhammadahmadasifbhatti We need to keep `feat: add agoric-upgrade-19 (proposal 91)` as single feat commit instead of `init` (might need to create a separate PR with the right commit).
Also can we look at failure step? What is it about? We might need help from Mathieu or Mujahid for that.

mujahidkay commented:
> @muhammadahmadasifbhatti We need to keep `feat: add agoric-upgrade-19 (proposal 91)` as single feat commit instead of `init` (might need to create a separate PR with the right commit).

Can just `reword` commit. 

> Also can we look at failure step? What is it about? We might need help from Mathieu or Mujahid for that. 

@muhammadahmadasifbhatti we also need to copy the patches folder from a3p-integration

Muneeb147 commented:
<img width="1380" alt="Screenshot 2025-04-04 at 7 32 11 PM" src="https://github.com/user-attachments/assets/4476ee31-3c30-4d86-95a0-ed36011ddbe4" />

As mujahid said, we might need to copy patches folder as well. The error is on patch

mujahidkay commented on proposals/91:upgrade-19/README.md:
This README needs to be updated. It usually contains info regarding when the proposal was adopted on the chain i.e. the block height and its respective block time (see previous proposals for ref)

Muneeb147 commented:
@muhammadahmadasifbhatti Can you please share more details? Which test exactly? What is the exact error? Is it consistently failing at one single point?

muhammadahmadasifbhatti commented:
@Muneeb147 
> Can you please share more details? Which test exactly?

I have attached the links to the failed tests

> What is the exact error?

Exact error on the console is:
```
console.error(`Run of ${name} failed with exit code ${runResult.status}`);
                                         ^
ReferenceError: name is not defined
```
But this is not the cause. Seems like some directories are missing that should have been created when we ran `script/buill-all-submissions.sh`. 

> Is it consistently failing at one single point?

Yes, it is consitently failing at the same point.

Muneeb147 commented:
@muhammadahmadasifbhatti 
So pasting the exact test/error in the comment helps as the viewer can directly see the root-cause rather than having a friction of jumping to the links and finding the error.

Muneeb147 commented:
@muhammadahmadasifbhatti 

```
Error: axios@patch:axios@npm%3A1.7.7#~/.yarn/patches/axios-npm-1.7.7-cfbedc233d.patch: ENOENT: no such file or directory, open '/usr/src/proposals/91:upgrade-19/.yarn/patches/axios-npm-1.7.7-cfbedc233d.patch'
```
Why are we seeing this error again? I see you have added patches folder in recent commit. Landed on this error from your `Some tests` link.

muhammadahmadasifbhatti commented:
@Muneeb147 Wrong Link was attached. Updated it. 

**New update**: The issue is with the `scripts/build-all-submissions.sh`. Ideally it should create the directories given in the `sdk-generate` in the `package.json` but, that's not the case. 

I re-ran the `scripts/build-all-submissions.sh` and got this error. 
```
#  node[89177]: void node::fs::InternalModuleStat(const FunctionCallbackInfo<Value> &) at ../src/node_file.cc:1056
  #  Assertion failed: (args.Length()) >= (2)

----- Native stack trace -----

 1: 0x102350840 node::Assert(node::AssertionInfo const&) [/opt/homebrew/Cellar/node/23.11.0/bin/node]
 2: 0x10236507c node::fs::InternalModuleStat(v8::FunctionCallbackInfo<v8::Value> const&) [/opt/homebrew/Cellar/node/23.11.0/bin/node]
 ...
```

Looking further into it.

Muneeb147 commented:
@muhammadahmadasifbhatti I see all green. That's great Ahmad ❤️❤️
So how did you figure out the problem and the solution eventually? Do we need to update docs as well for future reference?

mujahidkay reviewed (approved):
Please mark the relevant checkboxes as complete, and the remaining once you open the companion PR.

muhammadahmadasifbhatti commented:
> @muhammadahmadasifbhatti I see all green. That's great Ahmad ❤️❤️ So how did you figure out the problem and the solution eventually? Do we need to update docs as well for future reference?

I was not able to generate the directories mentioned in the `sdk-generate` in `package.json` due to a `node` error. Actually, I changed my machine recently, and it was using the latest `node version`. I checked the `agorik-sdk` docs and they state
> We generally support latest version 

So, I changed the version to the minimum version supported by `agorik-sdk` and got the `sdk-generate` directories.

Chris-Hibbert commented:
I'm currently working on [adding the fastUSDC coreEvals (proposals 87 and 88)](https://github.com/Agoric/agoric-private/issues/305) to `agoric-3-proposals`.  Please don't remove the current contents of `n:upgrade-next` and `g:gtm-fast-usdc` from a3p-integration until we have all these migrated.
